src/mesh.cpp

```diff
@@ -2324,16 +2324,49 @@ const std::string LibMesh::mesh_lib_type = "libmesh";
 
 LibMesh::LibMesh(pugi::xml_node node) : UnstructuredMesh(node)
 {
+  std::string filename = get_node_value(node, "filename");
+  double length_multiplier = std::stod(get_node_value(node, "length_multiplier"));
+  set_mesh_pointer_from_filename(filename);
+  set_length_multiplier(length_multiplier);
+  if (specified_length_multiplier_) {
+    libMesh::MeshTools::Modification::scale(*m_, length_multiplier_);
+  }
+  m_->prepare_for_use();
```

Comment:
That is a good point. Maybe something like `scale_and_prepare_for_use()`. Since `specified_length_multiplier_`,`m_`, and `length_multiplier_` are member variables that get set before these lines, I think this method could take no parameters.

Reply:
I don't think we want to call `prepare_for_use()` for the mesh that MOOSE will be passing in. MOOSE already handles calling this function for us, which does things like [partition the mesh, finds element neighbor connectivity, etc](https://libmesh.github.io/doxygen/classlibMesh_1_1MeshBase.html#:~:text=Prepare%20a%20newly,set%20allow_remote_element_removal(false).). 

But perhaps @pshriwise could confirm?

Reply:
You're right @aprilnovak. I don't think we'll want to call `prepare_for_use` here if the mesh is being managed by MOOSE. 

Here's what I'll propose to move this into the `LibMesh::initialize` method for code reuse, but avoid a misplaced `prepare_for_use` call. If the mesh isn't managed by OpenMC, the `LibMesh::unique_m_` pointer will be null, which we can use to indicate whether or not OpenMC is managing the mesh. At the beginning of the `initialize` method we could have:

```cpp
// if OpenMC is managing the libMesh::MeshBase instance, prepare the mesh.
// Otherwise assume that it is prepared by it's owning application
if (!unique_m_) {
   m_->prepare_for_use();
}
```
Thoughts?

Reply:
Another thing we should do is check that the mesh is prepared before transport begins, but that can be done in a separate PR. I'll make an issue...

Reply:
I'll also note that it looks to me like the the `set_length_multiplier` calls can be moved into the `initialize` method using this approach.

Reply:
I like the unique ptr access approach - we may need to explicitly initialize it to nullptr in the header file for the "MOOSE providing mesh" case, I think some compilers might not default to null initialization if unspecified.

Reply:
Good ideal 👍🏻

Reply:
I can add this ` unique_ptr<libMesh::MeshBase> unique_m_ = nullptr; //!< pointer to the libMesh MeshBase instance, only used if mesh is created inside OpenMC`. Should we leave `m_` alone or set it to `nullptr` as well for consistency? @aprilnovak